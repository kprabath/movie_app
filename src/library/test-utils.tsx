import {NavigationContainer} from '@react-navigation/native';
import {render} from '@testing-library/react-native';
import React, {ReactElement} from 'react';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import ThemeProvider from '../contexts/theme-context';
import {store} from '../store';
import {ThemeA} from '../theme/themes';

export const renderWithRedux = (component: ReactElement) => {
  return {
    ...render(
      <ThemeProvider value={{theme: ThemeA, setTheme: jest.fn()}}>
        <SafeAreaProvider
          initialSafeAreaInsets={{top: 1, left: 2, right: 3, bottom: 4}}>
          <Provider store={store}>{component}</Provider>
        </SafeAreaProvider>
      </ThemeProvider>,
    ),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
};

export const renderWithReduxAndNavi = (component: ReactElement) => {
  const App = component;

  return {
    ...render(
      <Provider store={store}>
        <NavigationContainer>
          <App />
        </NavigationContainer>
      </Provider>,
    ),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
};

export const renderWithMockStore = (
  component: ReactElement,
  mockStore: object,
) => {
  const mockReducer = configureStore([]);
  const store = mockReducer(mockStore);

  return {
    ...render(
      <SafeAreaProvider
        initialSafeAreaInsets={{top: 1, left: 2, right: 3, bottom: 4}}>
        <Provider store={store}>{component}</Provider>
      </SafeAreaProvider>,
    ),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
};

export const renderWithSafeArea = (component: ReactElement) => ({
  ...render(
    <ThemeProvider value={{theme: ThemeA, setTheme: jest.fn()}}>
      <SafeAreaProvider
        initialSafeAreaInsets={{top: 1, left: 2, right: 3, bottom: 4}}>
        {component}
      </SafeAreaProvider>
    </ThemeProvider>,
  ),
});

export const makeProps = (
  params: {[key: string]: any} = {},
  props?: object,
) => ({
  navigation: {
    addListener: jest.fn().mockReturnValue(jest.fn),
    navigate: jest.fn(),
    goBack: jest.fn(),
    reset: jest.fn(),
    popToTop: jest.fn(),
    push: jest.fn(),
  },
  route: {params},
  ...props,
});
