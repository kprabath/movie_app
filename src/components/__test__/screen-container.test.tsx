import {makeProps, renderWithSafeArea} from '../../library/test-utils.tsx';
import ScreenContainer from '../screen-container/screen-container';

describe('ScreenContainer', () => {
  test('should match the snapshot', () => {
    const props = makeProps();
    const componet = renderWithSafeArea(<ScreenContainer {...props} />);
    expect(componet).toMatchSnapshot();
  });
});
