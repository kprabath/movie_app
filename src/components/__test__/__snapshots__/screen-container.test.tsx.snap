// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenContainer should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <BVLinearGradient
    colors={
      Array [
        4281282869,
        4279571487,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 4,
          "paddingTop": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 100,
            "paddingHorizontal": 16.666666666666668,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "maxWidth": 187.5,
            },
            Object {
              "flexDirection": "row",
            },
            Object {
              "zIndex": 10,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "paddingHorizontal": 16.666666666666668,
              },
              undefined,
              Object {
                "backgroundColor": "#e74c3c",
                "borderRadius": 40,
                "padding": 16.666666666666668,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "maxWidth": 375,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "fontSize": 23.333333333333336,
                },
                Array [
                  Object {
                    "fontSize": 40,
                  },
                  undefined,
                ],
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "maxWidth": 187.5,
            },
            Object {
              "justifyContent": "flex-start",
            },
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "zIndex": 10,
            },
            undefined,
          ]
        }
      />
    </View>
    <RCTScrollView
      contentContainerStyle={
        Array [
          undefined,
        ]
      }
      showsVerticalScrollIndicator={false}
    >
      <View />
    </RCTScrollView>
  </BVLinearGradient>
</RNCSafeAreaProvider>
`;
